src/EnergyPlus/UnitarySystem.cc

```diff
@@ -11608,6 +11360,15 @@ namespace UnitarySystems {
                 LatentLoad = false;
             }
 
+            // disable waterside economizer if the condition is NOT favorable
+            if (this->m_waterSideEconomizerFlag) {
+                if (!this->runWaterSideEconomizer) {
+                    SensibleLoad = false;
+                    LatentLoad = false;
+                    HXUnitOn = false;
+                }
+            }
+
```

Comment:
If the new coilsystem water object is used as water-side economizer but the condition is not favorable  to operate then the coilsystem is disabled by setting these local variables to FALSE.